### PR TITLE
TST: filter out factorial(float) deprecation warning

### PR DESCRIPTION
**Pull request by rgommers**
This was causing the python3.9-dev CI to fail.

Also add a missing file to `.gitignore`.

### CODE REVIEW

File: scipy/special/tests/test_basic.py [hunk 1]
```diff
@@ -1824,10 +1824,12 @@ def test_nan_inputs(self, x, exact):
 
     def test_mixed_nan_inputs(self):
         x = np.array([np.nan, 1, 2, 3, np.nan])
-        result = special.factorial(x, exact=True)
-        assert_equal(np.array([np.nan, 1, 2, 6, np.nan]), result)
-        result = special.factorial(x, exact=False)
-        assert_equal(np.array([np.nan, 1, 2, 6, np.nan]), result)
+        with suppress_warnings() as sup:
+            sup.filter(DeprecationWarning, "Using factorial() with floats is deprecated")
```
Comment (pv) [inline scipy/special/tests/test_basic.py:1828]: Does this need to be `, match="..."` to work?

Comment (pv) [inline scipy/special/tests/test_basic.py:1828]: Does this need `\\(\\)`?

Comment (rgommers) [reply scipy/special/tests/test_basic.py]: Looks like it. Didn't test locally; this warning only showed up in 3.9-dev so I assume something is changing in deprecation warnings default behavior.

**Comment by rgommers** [comment]
Fixes CI and is a trivial change, so I'll merge it.

**Pull request merged**